src/components/fields/MultiColorPicker.js

```diff
@@ -0,0 +1,166 @@
+import React, {Component} from 'react';
+import Color from './Color';
+import Colorscale from './Colorscale';
+import Info from './Info';
+import Field from './Field';
+import RadioBlocks from '../widgets/RadioBlocks';
+import PropTypes from 'prop-types';
+import {adjustColorscale, connectToContainer} from 'lib';
+
+class UnconnectedMultiColorPicker extends Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      selectedConstantColorOption:
+        context.traceIndexes.length > 1 &&
+        props.fullValue.every(v => v[1] === props.fullValue[0][1])
+          ? 'single'
+          : 'multiple',
+    };
+    this.setColor = this.setColor.bind(this);
+    this.setColors = this.setColors.bind(this);
+  }
+
+  setColor(color) {
+    this.props.updatePlot(color);
```

Comment:
why not just pass updatePlot as a prop?

Reply:
It is passed, from the connect wrapper, I just prefer having a separate function, 
one setColor, one setColors, that's all